### PR TITLE
mod bug fix

### PR DESCRIPTION
**Pull request by mjafaari**

### CODE REVIEW

**Comment by mwoss** [comment]
Hi! Could you provide new tests for your change? Current ones were not enough to cover edge cases.

File: decimal.go [hunk 1]
```diff
@@ -610,8 +610,11 @@ func (d Decimal) DivRound(d2 Decimal, precision int32) Decimal {
 
 // Mod returns d % d2.
 func (d Decimal) Mod(d2 Decimal) Decimal {
-	quo := d.Div(d2).Truncate(0)
-	return d.Sub(d2.Mul(quo))
+	res := d
+	for res.Cmp(d2) > 0 {
+		res = res.Sub(d2)
+	}
```
Comment (mwoss) [inline decimal.go:616]: This would fail on basic cases like 7 mod 2, also it does not support negative numbers

**Review by mwoss — CHANGES_REQUESTED** [review]

**Comment by mwoss** [comment]
Issue fixed in #258 and then in #317. Closing.
